vault/resource_database_secret_backend_static_role.go

```diff
@@ -78,43 +97,52 @@ func databaseSecretBackendStaticRoleResource() *schema.Resource {
 	}
 }
 
-func databaseSecretBackendStaticRoleWrite(d *schema.ResourceData, meta interface{}) error {
+func databaseSecretBackendStaticRoleWrite(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, e := provider.GetClient(d, meta)
 	if e != nil {
-		return e
+		return diag.FromErr(e)
 	}
 
-	backend := d.Get("backend").(string)
-	name := d.Get("name").(string)
+	backend := d.Get(consts.FieldBackend).(string)
+	name := d.Get(consts.FieldName).(string)
 
 	path := databaseSecretBackendStaticRolePath(backend, name)
 
+	useAPIVer115 := provider.IsAPISupported(meta, provider.VaultVersion115)
+	if useAPIVer115 {
+		if d.HasChange(consts.FieldRotationSchedule) {
+			staticRoleFields = append(staticRoleFields, "rotation_schedule")
```

Comment:
Suggest using the constants for these fields here :) 
```suggestion
			staticRoleFields = append(staticRoleFields, consts.FieldRotationSchedule)
```